cloudify_ansible/tasks.py

```diff
@@ -271,9 +289,55 @@ def ansible_remove_host(new_sources_dict, _ctx, **_):
 @operation
 @prepare_ansible_node
 def precreate(ctx=None, **_):
+    _precreate(ctx, **_)
+
+
+def _precreate(ctx=None, **_):
     ctx.logger.info('Checking Ansible installation.')
     if not utils.get_instance().runtime_properties.get(
             constants.PLAYBOOK_VENV):
         ctx.logger.error('Ansible will need to be installed. '
                          'This may cause problems if Ansible is installed in '
                          'a relationship operation.')
+
+
+@operation
+def install(ctx=None, **_):
+    install_config = ctx.node.properties
+    utils.create_playbook_venv(ctx)
+    utils.create_playbook_workspace(ctx)
+    utils.install_extra_packages(
+        ctx,
+        install_config.get('extra_packages')
+    )
+    utils.install_galaxy_collections(
+        ctx,
+        install_config.get('galaxy_collections')
+    )
+
+
+@operation
+def uninstall(ctx=None, **_):
+    if ctx.node.properties.get('ansible_external_pyenv'):
+        ctx.logger.info("Not deleting external pyenv")
+        return
+    if ctx.node.properties.get('ansible_external_executable_path'):
+        ctx.logger.info("Not deleting external executable path")
+        return
+    utils.delete_playbook_environment(ctx)
+    utils.delete_playbook_workspace(ctx)
+
+
+@operation
+def set_pyenv(playbook_venv, ctx, **_):
```

Comment:
let's rename to `set_venv`. Since, these are virtualenvs not pyenvs.

Reply:
I'm going to delete it since the relationship does not make sense anymore